ci/compile.sh

```diff
@@ -8,9 +8,8 @@ source "${WORKDIR}"/ci/archs.sh
 
 # Since race detector has huge performance price and it works only on amd64 and does not
 # work with pie executables, its enabled only for development builds.
-branch="${CI_COMMIT_REF_NAME:=$(git describe --contains --all HEAD)}"
```

Comment:
Firstly I tried fixing this by adding `--git-dir=${WORKDIR}` to the git command. Somehow it didn't work and then I thought whether we need this branch condition altogether. We have clearly separated dev and candidate builds now, so building differently based on source branch is just confusing (as proved by the issue we are having). So I decided to simply remove that.